exchanges/hitbtc/hitbtc_websocket.go

```diff
@@ -72,50 +72,83 @@ func (h *HitBTC) WsHandleData() {
 			}
 			h.Websocket.TrafficAlert <- struct{}{}
 
-			var init capture
-			err = json.Unmarshal(resp.Raw, &init)
+			err = h.wsHandleData(resp.Raw)
 			if err != nil {
 				h.Websocket.DataHandler <- err
-				continue
-			}
-			if init.Error.Code == 1002 {
-				h.Websocket.SetCanUseAuthenticatedEndpoints(false)
-			}
-			if init.ID > 0 {
-				h.WebsocketConn.AddResponseWithID(init.ID, resp.Raw)
-				continue
 			}
-			if init.Error.Message != "" || init.Error.Code != 0 {
-				h.Websocket.DataHandler <- fmt.Errorf("hitbtc.go error - Code: %d, Message: %s",
-					init.Error.Code,
-					init.Error.Message)
-				continue
-			}
-			if _, ok := init.Result.(bool); ok {
-				continue
-			}
-			if init.Method != "" {
-				h.handleSubscriptionUpdates(resp, init)
-			} else {
-				h.handleCommandResponses(resp, init)
+		}
+	}
+}
+
+func (h *HitBTC) wsGetTableName(respRaw []byte) (string, error) {
+	var init capture
+	err := json.Unmarshal(respRaw, &init)
+	if err != nil {
+		return "", err
+	}
+	if init.Error.Code == 1002 {
```

Comment:
Please make a const type for this (AuthFailure) to prevent the use of magic numbers